### ISSUE TITLE
allow `type(self)._private_attribute`

### ISSUE THREAD

**Issue by anntzer** [PyCQA/pylint#1031]
### Steps to reproduce

pylint the following

```
class C:
    _attr = 1

    def foo(self):
        return type(self)._attr
```
### Current behavior

A warning is raised: `W:  5,15: Access to a protected member _attr of a client class (protected-access)`
### Expected behavior

No warning raised.  After all, I can already access `_attr` as `self._attr`, and `type(self)._attr` simply allows me to clarify that I am indeed accessing a class attribute, not an instance attribute.
### pylint --version output

pylint 1.6.4, 
astroid 1.4.7
Python 3.5.2 (default, Jun 28 2016, 08:46:01) 
[GCC 6.1.1 20160602]

**Comment by rogalski** [comment]
Requires extra if in https://github.com/PyCQA/pylint/blob/master/pylint/checkers/classes.py#L927
Should be very similar to `super()` case.
